src/garage/torch/algos/ppo.py

```diff
@@ -0,0 +1,96 @@
+            policies.
+        n_samples (int): Number of train_once calls per epoch.
+        discount (float): Discount.
+        gae_lambda (float): Lambda used for generalized advantage
+            estimation.
+        center_adv (bool): Whether to rescale the advantages
+            so that they have mean 0 and standard deviation 1.
+        positive_adv (bool): Whether to shift the advantages
+            so that they are always positive. When used in
+            conjunction with center_adv the advantages will be
+            standardized before shifting.
+        optimizer (object): The optimizer of the algorithm. Should be the
+            optimizers in torch.optim.
+        policy_ent_coeff (float): The coefficient of the policy entropy.
+            Setting it to zero would mean no entropy regularization.
+        use_softplus_entropy (bool): Whether to estimate the softmax
+            distribution of the entropy to prevent the entropy from being
+            negative.
+        stop_entropy_gradient (bool): Whether to stop the entropy gradient.
+        entropy_method (str): A string from: 'max', 'regularized',
+            'no_entropy'. The type of entropy method to use. 'max' adds the
+            dense entropy to the reward for each time step. 'regularized' adds
+            the mean entropy to the surrogate objective. See
+            https://arxiv.org/abs/1805.00909 for more details.
+    """
+
+    def __init__(self,
+                 env_spec,
+                 policy,
+                 baseline,
+                 max_path_length=500,
+                 policy_lr=1e-2,
+                 lr_clip_range=1e-2,
+                 n_samples=1,
+                 discount=0.99,
+                 gae_lambda=1,
+                 center_adv=True,
+                 positive_adv=False,
+                 optimizer=None,
+                 policy_ent_coeff=0.0,
+                 use_softplus_entropy=False,
+                 stop_entropy_gradient=False,
+                 entropy_method='no_entropy'):
+        super().__init__(env_spec, policy, baseline, max_path_length,
+                         policy_lr, n_samples, discount, gae_lambda,
+                         center_adv, positive_adv, optimizer, policy_ent_coeff,
+                         use_softplus_entropy, stop_entropy_gradient,
+                         entropy_method)
+
+        self._constraint_clip_range = lr_clip_range
+        self._old_policy_state_dict = self.policy.state_dict()
+
+    def _compute_objective(self, advantages, obs, actions, rewards):
+        # Compute constraint
+        current_policy_state_dict = self.policy.state_dict()
+        self.policy.load_state_dict(self._old_policy_state_dict)
+
+        with torch.no_grad():
+            old_ll = self.policy.log_likelihood(obs, actions)
+
+        self.policy.load_state_dict(current_policy_state_dict)
+        new_ll = self.policy.log_likelihood(obs, actions)
+
+        constraint = (new_ll - old_ll).exp()
```

Comment:
usually the word "constraint" refers to the rule used to constrain a value, not the constrained value itself.

this isn't a constraint, it's a likelihood ratio. we constrain the likelihood ratio by clipping it later, so in this case the constraint is lr_clip_range. please name this more clearly.

Reply:
changed the name into likelihood_ratio

Reply:
i don't see the new name -- did you push to origin yet?